### PR TITLE
hints added to maximum functions

### PR DESCRIPTION
**Pull request by shubham-shay-te**

### CODE REVIEW

File: core_lib/cache/cache_handler_ram.py [hunk 1]
```diff
@@ -8,7 +8,7 @@ class CacheHandlerRam(CacheHandler):
     def __init__(self):
         self.cached_function_responses = {}
 
-    def get(self, key):
+    def get(self, key) -> None:
```
[the remark this answers is not in the document]
Comment (shubham-shay-te) [reply core_lib/cache/cache_handler_ram.py:11]: here I had `dict` earlier but pycharm suggested to change it to `None`

File: core_lib/data_layers/data/handler/sql_alchemy_data_handler_registry.py [hunk 2]
```diff
@@ -20,11 +20,11 @@ def __init__(self, config: DictConfig):
             Base.metadata.create_all(self._engine)
 
     @property
-    def engine(self):
+    def engine(self) -> create_engine:
```
[the remark this answers is not in the document]
Comment (shubham-shay-te) [reply core_lib/data_layers/data/handler/sql_alchemy_data_handler_registry.py:23]: it's returning `engine`

File: core_lib/cache/cache_handler_ram.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shay-te) [reply core_lib/cache/cache_handler_ram.py:11]: thank you for noticing. it coult be None, dict, int, string. anything. so maybe better to remove this one

File: core_lib/data_layers/data/handler/sql_alchemy_data_handler_registry.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def _on_db_session_exit(self, db_session: SqlAlchemyDataHandler):
         else:
             db_session.close()
 
-    def _create_engine(self, config):
+    def _create_engine(self, config) -> create_engine:
```
Comment (shay-te) [inline core_lib/data_layers/data/handler/sql_alchemy_data_handler_registry.py:57]: this as rell return `engine`

another change please. 
just return the `create_engine` statment. variable `engine` not in use

File: core_lib/helpers/datetime_utils.py [hunk 4]
```diff
@@ -22,112 +24,112 @@ def _next_weekday(weekday, hours: int = 0, minutes: int = 0):
...
+def year_end(hours: int = 0, minutes: int = 0) -> datetime:
     return year_begin(hours=hours, minutes=minutes) + relativedelta(years=1)
 
 
 #
 # UTILS
 #
 
-def age(born: date):
+def age(born: date) -> Union[int, bool]:
```
Comment (shay-te) [inline core_lib/helpers/datetime_utils.py:129]: age should return only a number. when it returns a bool?

File: core_lib/web_helpers/web_helprs_utils.py [hunk 5]
```diff
@@ -14,7 +14,7 @@ def init(server_type: ServerType):
         WebHelpersUtils.server_type = server_type
 
     @staticmethod
-    def get_server_type():
+    def get_server_type() -> server_type:
```
Comment (shay-te) [inline core_lib/web_helpers/web_helprs_utils.py:17]: did you mean to return `WebHelpersUtils.ServerType` instead?

what kind of type is it `server_type`?

**Review by shay-te — CHANGES_REQUESTED** [review]

File: core_lib/web_helpers/web_helprs_utils.py (hunk 5 above)
Comment (shubham-shay-te) [reply core_lib/web_helpers/web_helprs_utils.py:17]: it is returning the `server_type`  which will have the actual value, but `ServerType` is the enum class that's why i went with `server_type`

File: core_lib/cache/cache_decorator.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import logging
 from datetime import timedelta, datetime, timezone
 from functools import wraps
-from typing import Union
+from typing import Union, Any
```
Comment (shay-te) [inline core_lib/cache/cache_decorator.py:4]: Who ius using `Union` and `Any`?

File: core_lib/data_layers/data/handler/sql_alchemy_data_handler_registry.py [hunk 7]
```diff
@@ -54,7 +54,7 @@ def _on_db_session_exit(self, db_session: SqlAlchemyDataHandler):
         else:
             db_session.close()
 
-    def _create_engine(self, config):
+    def _create_engine(self, config) -> create_engine:
         engine = create_engine(build_url(**config.url),
                                pool_recycle=config.session.pool_recycle,
                                echo=config.log_queries)
```
Comment (shay-te) [inline core_lib/data_layers/data/handler/sql_alchemy_data_handler_registry.py:60]: sorry my previus comment was not clear 
- there is no need for the `engine` variable, just return from the funciton it self 
- the function return type `engine` 

```
def _create_engine(self, config) -> engine:
    return create_engine(build_url(**config.url),
                                pool_recycle=config.session.pool_recycle,
                                echo=config.log_queries)
```

File: core_lib/data_transform/result_to_dict.py [hunk 8]
```diff
@@ -2,7 +2,7 @@
 import enum
 from collections import Iterable
 from functools import wraps
-from typing import Callable, Awaitable
+from typing import Callable, Awaitable, Union, List, Any
```
Comment (shay-te) [inline core_lib/data_transform/result_to_dict.py:5]: all of the import are being used?

**Review by shay-te — COMMENTED** [review]
Good work, please let visit the comments

File: core_lib/cache/cache_decorator.py (hunk 6 above)
Comment (shubham-shay-te) [reply core_lib/cache/cache_decorator.py:4]: removed `Any`, `Union` is being used in the `init` for `expire: Union[timedelta, str] = None`

File: core_lib/data_layers/data/handler/sql_alchemy_data_handler_registry.py (hunk 7 above)
Comment (shubham-shay-te) [reply core_lib/data_layers/data/handler/sql_alchemy_data_handler_registry.py:60]: done

File: core_lib/data_transform/result_to_dict.py (hunk 8 above)
Comment (shubham-shay-te) [reply core_lib/data_transform/result_to_dict.py:5]: done

**Review by shay-te — APPROVED** [review]

**Pull request merged**